### PR TITLE
Translate the Launchpad

### PR DESCRIPTION
**Pull request by samajammin**
Add translation support!

While going through the process of configuring the codebase to support translations, we ended up making some significant design & content changes to the app 😅 

Fixes https://github.com/ethereum/eth2.0-deposit/issues/302

### CODE REVIEW

**Comment by samajammin** [comment]
FYI @ryancreatescopy @wackerow - I've just rebased this branch of `dev`, so we should have all those changes from #300 that Carl merged.

**Comment by samajammin** [comment]
@CarlBeek I believe we've finished everything 🎉! This PR is now ready for review.

cc @wackerow @ryancreatescopy

File: src/pages/Transactions/index.tsx [hunk 1]
```diff
@@ -85,27 +87,39 @@ const _TransactionsPage = ({
...
   const createButtonText = (): string => {
     if (totalTxCount === 1) {
-      return 'Initiate the transaction.';
+      return formatMessage({ defaultMessage: 'Confirm deposit' });
     }
     if (totalTxCount === remainingTxCount && totalTxCount > 0) {
-      return `Initiate all ${totalTxCount} transactions`;
+      return formatMessage(
+        {
+          defaultMessage: `Confirm all {totalTxCount} deposits`,
```
Comment (hwwhww) [inline src/pages/Transactions/index.tsx:95]: nitpicking: using the word "confirm" may sound like "waiting for 12 block confirmations...". How about using "send"?

File: src/pages/Transactions/index.tsx [hunk 2]
```diff
@@ -134,22 +148,35 @@ const _TransactionsPage = ({
...
-        <Text className="mt20">
-          You must sign an individual transaction for each key you created.
-        </Text>
-        <Text className="mt10">
-          You can initiate these all at once, or sign them individually from the
-          key-list below
-        </Text>
+        {depositKeys.length === 1 && (
+          <Text className="mt20">
+            <FormattedMessage defaultMessage="Submit a transaction to confirm your deposit." />
```
Comment (hwwhww) [inline src/pages/Transactions/index.tsx:163]: ditto

```suggestion
            <FormattedMessage defaultMessage="Submit a transaction to finish your deposit." />
```

File: src/pages/TopUp/components/ValidatorTable.tsx [hunk 3]
```diff
@@ -101,62 +121,79 @@ const ValidatorTable: React.FC<{
 
   const validatorRows = React.useMemo(() => {
     return validators.map(validator => {
-      const alreadyToppedUp = validator.effectivebalance >= 32000000000;
-      // const statusIneligible =
-      //   validator.status === 'slashed' || validator.status === 'exited';
-
-      // const disableTopUpButton = alreadyToppedUp || statusIneligible;
+      const alreadyToppedUp =
+        validator.effectivebalance >= Number(PRICE_PER_VALIDATOR) * 10 ** 9;
```
Comment (hwwhww) [inline src/pages/TopUp/components/ValidatorTable.tsx:125]: nitpick: could use envVars `ETHER_TO_GWEI`.

File: src/pages/TopUp/components/ValidatorTable.tsx [hunk 4]
```diff
@@ -101,62 +121,79 @@ const ValidatorTable: React.FC<{
...
+                  <FormattedMessage defaultMessage="YES" />
+                ) : (
+                  <FormattedMessage defaultMessage="NO" />
+                )}
+              </Text>
             </TableCell>
             <TableCell scope="col" border="bottom">
               <Text>
-                {numeral(validator.balance / 10 ** 9).format('0.00000')} ETH
+                {`${numeral(validator.balance / 10 ** 9).format(
```
Comment (hwwhww) [inline src/pages/TopUp/components/ValidatorTable.tsx:172]: ditto. could use envVars `ETHER_TO_GWEI`.

File: src/pages/TopUp/components/ValidatorTable.tsx [hunk 5]
```diff
@@ -101,62 +121,79 @@ const ValidatorTable: React.FC<{
...
+                  '0.00000'
+                )} ${TICKER_NAME}`}
               </Text>
             </TableCell>
             <TableCell scope="col" border="bottom">
               <Text>
-                {numeral(validator.effectivebalance / 10 ** 9).format(
+                {`${numeral(validator.effectivebalance / 10 ** 9).format(
```
Comment (hwwhww) [inline src/pages/TopUp/components/ValidatorTable.tsx:179]: ditto

**Review by hwwhww — COMMENTED** [review]
Epic milestone! 🎉

Sorry that I didn't follow the previous PRs so I might miss something. Tried to scan every file and left some comments.

Thank y'all for the great work! 🔥

File: src/utils/envVars.ts [hunk 6]
```diff
@@ -44,6 +47,8 @@ if(process.env.REACT_APP_PRICE_PER_VALIDATOR && Number.isNaN(Number(process.env.
 }
 export const PRICE_PER_VALIDATOR        = process.env.REACT_APP_PRICE_PER_VALIDATOR || 32;
 
+export const EJECTION_PRICE = 16;
```
Comment (hwwhww) [inline src/utils/envVars.ts:50]: Maybe make it configurable as `PRICE_PER_VALIDATOR`?

```suggestion
if(process.env.REACT_APP_EJECTION_PRICE && Number.isNaN(Number(process.env.REACT_APP_EJECTION_PRICE))) {
    throw new Error("REACT_APP_EJECTION_PRICE must be of type: number")
}
export const EJECTION_PRICE             = process.env.REACT_APP_EJECTION_PRICE || 16;
```

File: src/components/AppBar.tsx [hunk 7]
```diff
@@ -112,39 +161,48 @@ const _AppBar = ({ location }: RouteComponentProps) => {
...
+                active={pathname === routesEnum.landingPage}
+                level={4}
+                margin="none"
+                className="bar-link-text no-padding"
+              >
+                <Text>
+                  <FormattedMessage
+                    defaultMessage="Eth2 {network} Launchpad"
+                    values={{
+                      network: IS_MAINNET ? '' : `${TESTNET_LAUNCHPAD_NAME} `,
```
[the remark this answers is not in the document]
Comment (wackerow) [reply src/components/AppBar.tsx:178]: The idea here was to show "Eth2 Launchpad" for mainnet (leaving out the word 'mainnet') and to show "Eth2 Pyrmont Launchpad" for Pyrmont testnet.

File: src/pages/Clients/Eth1/Besu.tsx [hunk 8]
```diff
@@ -4,80 +4,93 @@ import { IS_MAINNET } from '../../../utils/envVars';
...
+        <FormattedMessage
+          defaultMessage="Use {http} to connect your Eth2 node to the JSON RPC
+          endpoint. This will enable the JSON RPC services on the default 8545
+          port."
+          values={{
+            http: <Code className="mt20">-rpc-http-enabled</Code>,
+          }}
+          description="{http} shows '-rpc-http-enabled' terminal command"
+        />
       </Text>
```
[the remark this answers is not in the document]
Comment (wackerow) [reply src/pages/Clients/Eth1/Besu.tsx:86]: will patch

File: src/pages/Landing/Hero.tsx [hunk 9]
```diff
@@ -142,14 +148,20 @@ export const Hero = () => {
...
-                        <EthLogo src={EthDiamondPlain} />
                         <LogoText>
-                          Eth2 Launch Pad{' '}
-                          {IS_MAINNET ? `` : `for ${ETH2_NETWORK_NAME} testnet`}
+                          {IS_MAINNET ? (
+                            <FormattedMessage defaultMessage="Eth2 Launchpad" />
+                          ) : (
+                            <FormattedMessage
+                              defaultMessage="Eth2 Launchpad for {TESTNET_LAUNCHPAD_NAME} testnet"
+                              values={{ TESTNET_LAUNCHPAD_NAME }}
```
[the remark this answers is not in the document]
Comment (wackerow) [reply src/pages/Landing/Hero.tsx:161]: Again, this will only show up if user is on testnet, in which case `ETH2_NETWORK_NAME === TESTNET_LAUNCHPAD_NAME`... 
`TESTNET_LAUNCHPAD_NAME` is used here to emphasize that this will never say "mainnet"

File: src/pages/Landing/StakingRewardsChart/index.tsx [hunk 10]
```diff
@@ -131,30 +133,15 @@ const StakingRewardsChart: React.FC<Props> = ({
...
-                  >
-                    <text style={{ fontWeight: 'bold', fill: '#ADADAD' }}>
-                      {(524_288).toLocaleString()}
-                      &nbsp;{TICKER_NAME}
-                    </text>
-                    <text y={17} style={{ fill: '#ADADAD' }}>
-                      Required for Beacon Chain Launch
-                    </text>
-                  </g>
-
```
[the remark this answers is not in the document]
Comment (wackerow) [reply src/pages/Landing/StakingRewardsChart/index.tsx:157]: For the threshold to allow genesis we ended up deciding this was essentially no longer relevant given we are 3 months past genesis now and 6x the minimum needed. As for the total ETH staked at this time, this is shown both in the NetworkStatus component, and the StakingRewards component, both on the landing page.

File: src/pages/Phishing/index.tsx [hunk 11]
```diff
@@ -11,101 +14,139 @@ const SectionTitle = styled(Heading)`
...
         <ul>
           <li>
             <Text className="mt10">
-              Is the URL of the site you are making the deposit the one you
-              expect?
+              <FormattedMessage
+                defaultMessage="Does the site asking you for your {ethAmount} deposit have a URL you
+                expect?"
+                values={{
+                  ethAmount: `${PRICE_PER_VALIDATOR} ${TICKER_NAME}`,
```
[the remark this answers is not in the document]
Comment (wackerow) [reply src/pages/Phishing/index.tsx:39]: Yeah interesting point. May be worth breaking the mold for the phishing page, and just showing ETH even when user is in testnet mode. I can patch this.

File: src/pages/Transactions/index.tsx (hunk 1 above)
Comment (wackerow) [reply src/pages/Transactions/index.tsx:95]: @ryancreatescopy Thoughts on this? Could go either way... Perhaps "Sign all {} deposits"? It helps remove that feeling of "block confirmations". Agree it's a small issue

File: src/pages/TopUp/components/ValidatorTable.tsx (hunk 3 above)
Comment (wackerow) [reply src/pages/TopUp/components/ValidatorTable.tsx:125]: good call =) will patch

File: src/utils/envVars.ts [hunk 12]
```diff
@@ -1,4 +1,5 @@
 export const IS_MAINNET                 = Boolean(process.env.REACT_APP_IS_MAINNET !== 'false');  // If REACT_APP_IS_MAINNET is unset, set it to true by default
+export const TESTNET_LAUNCHPAD_NAME      = 'Pyrmont'
```
[the remark this answers is not in the document]
Comment (samajammin) [reply src/utils/envVars.ts:2]: Ya good catch, I should have changed this back after closing https://github.com/ethereum/eth2.0-deposit/pull/353

File: src/components/AppBar.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (samajammin) [reply src/components/AppBar.tsx:178]: Ya in this case it doesn't matter, `TESTNET_LAUNCHPAD_NAME` & `ETH2_NETWORK_NAME` will both be `Pyrmont`. Having at `ETH2_NETWORK_NAME` is prob a safe bet tho 👍

File: src/pages/Transactions/index.tsx (hunk 1 above)
Comment (samajammin) [reply src/pages/Transactions/index.tsx:95]: Agree use of "confirm" could be a bit confusing. I'm find with initiate or send.

File: src/pages/TopUp/components/ValidatorTable.tsx (hunk 3 above)
Comment (samajammin) [reply src/pages/TopUp/components/ValidatorTable.tsx:125]: Seems like a nice potential utility function 😃

File: src/components/AppBar.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (hwwhww) [reply src/components/AppBar.tsx:178]: @wackerow 
Sorry I meant replacing `TESTNET_LAUNCHPAD_NAME` with `ETH2_NETWORK_NAME` here. Just felt a bit strange here since we always use `ETH2_NETWORK_NAME` as "where am I". :)

@samajammin
Yeah IMO using `ETH2_NETWORK_NAME` would be more straightforward in this context.

File: .env.template [hunk 13]
```diff
@@ -9,6 +9,8 @@ REACT_APP_INFURA_PROJECT_ID=464bfe80298f4d0782f61h4b91de2dcb
 #
 # true, false
 REACT_APP_IS_MAINNET=true
+# string (only if REACT_APP_IS_MAINNET=false)
```
Comment (hwwhww) [inline .env.template:12]: Just to be clear, even if `REACT_APP_IS_MAINNET=true`, it could be set to an alternative testnet name (not the default string Prymont).

File: .env.mainnet [hunk 14]
```diff
@@ -1,5 +1,4 @@
 export REACT_APP_IS_MAINNET=true
 export REACT_APP_CONTRACT_ADDRESS=0x00000000219ab540356cBB839Cbe05303d7705Fa
 export REACT_APP_ETH_REQUIREMENT=524288
```
Comment (hwwhww) [inline .env.mainnet:3]: The .env files are for developers to switch between the network easily. You can type `source .env.pyrmont` or `source .env.mainnet` locally. I think it would be safer to add `REACT_APP_TESTNET_LAUNCHPAD_NAME`, so the future short-term or long-term testnet (*not* Pyrmont) envs won't overwrite it.

```suggestion
export REACT_APP_ETH_REQUIREMENT=524288
export REACT_APP_TESTNET_LAUNCHPAD_NAME=Pyrmont
```

**Review by hwwhww — APPROVED** [review]
The last nitpicking from me. 😅

File: src/pages/Acknowledgements/pageContent.tsx [hunk 15]
```diff
@@ -1,193 +1,289 @@
...
+      <>
+        <Text size="medium" className="mt10">
+          <FormattedMessage
+            defaultMessage="To become a validator you'll need to know about managing keys and
+              protecting a mnemonic. If you are not yet familiar with keys and mnemomics, please
+              do not proceed."
+          />
+        </Text>
+        <Text size="medium" className="mt20">
+          <FormattedMessage defaultMessage="We'll help you create a signing key and a withdrawal key for every validator you want to run. Signing and withdrawal keys are separate by design. Your signing key must be held online because it needs to be available at all times. Your withdrawal key doesn't need to be online so it's better to keep it separate for security." />
```
Comment (CarlBeek) [inline src/pages/Acknowledgements/pageContent.tsx:157]: I worry as it currently stands, this implies that a user will get their withdrawal keys now.
```suggestion
          <FormattedMessage defaultMessage="We'll help you create a signing key for every validator you want to run. Because there are no withdrawals until The Merge, you will not create your withdrawal keys now. When it is possible to withdraw you funds, you can derive your withdrawal keys from your mnemonic." />
```

File: src/pages/Acknowledgements/pageContent.tsx [hunk 16]
```diff
@@ -1,193 +1,289 @@
...
+        </Text>
+        <Text size="medium" className="mt20">
+          <FormattedMessage
+            defaultMessage="Both sets of keys will be generated
```
Comment (CarlBeek) [inline src/pages/Acknowledgements/pageContent.tsx:161]:
```suggestion
            defaultMessage="Your mnemonic is the backup for you signing keys and will be the ONLY way to
            withdraw your ETH when the time comes and no one can help you recover your mnemonic if
            you lose it."
```

File: src/pages/Acknowledgements/pageContent.tsx [hunk 17]
```diff
@@ -1,193 +1,289 @@
...
-      <Text size="large" className="my10">
-        Validator keys are derived from a unique mnemonic (or seed). Your seed
-        is the ONLY WAY to withdraw your funds. Above all, keep it safe!
-      </Text>
+      <>
+        <Text size="medium" className="mt10">
+          <FormattedMessage
+            defaultMessage="To become a validator you'll need to know about managing keys and
+              protecting a mnemonic. If you are not yet familiar with keys and mnemomics, please
+              do not proceed."
```
Comment (CarlBeek) [inline src/pages/Acknowledgements/pageContent.tsx:153]:
```suggestion
              do not proceed. Note that during this process you will not derive the keys needed to withdraw later, so store your mnemonic safely in order to be able to withdraw later."
```

File: src/pages/Acknowledgements/pageContent.tsx [hunk 18]
```diff
@@ -1,193 +1,289 @@
...
+    ),
     content: (
       <>
-        <Text size="large" className="my10">
-          Transfers between validators are disabled for now. Validators will
-          have to wait until <i>the merge</i> (around two years) to be able to
-          withdraw their ${TICKER_NAME}.
+        <Text size="medium" className="my10">
+          <FormattedMessage
+            defaultMessage={`Transfers between validators aren't possible today. You will have to
```
Comment (CarlBeek) [inline src/pages/Acknowledgements/pageContent.tsx:185]:
```suggestion
            defaultMessage={`Transfers between validators aren't possible yet. You will have to
```

File: src/pages/Acknowledgements/pageContent.tsx [hunk 19]
```diff
@@ -1,193 +1,289 @@
...
+              wait until mainnet merges with the Beacon Chain (around two years) before you can
+              withdraw your {TICKER_NAME}.`}
```
Comment (CarlBeek) [inline src/pages/Acknowledgements/pageContent.tsx:187]:
```suggestion
              withdraw or transfer your {TICKER_NAME}.`}
```

File: src/pages/Acknowledgements/pageContent.tsx [hunk 20]
```diff
@@ -1,193 +1,289 @@
...
+  [AcknowledgementIdsEnum.earlyAdoptionRisks]: {
+    title: <FormattedMessage defaultMessage="Early adopter risks" />,
     content: (
-      <Text size="large" className="my10">
-        With transfers disabled for now, there’s no way for a validator to
-        voluntarily exit and then restart later. This means validators need to
-        be in it for the long haul.
+      <Text size="medium" className="my10">
+        <FormattedMessage
+          defaultMessage="You're joining a young network. As with any new piece of software,
```
Comment (CarlBeek) [inline src/pages/Acknowledgements/pageContent.tsx:223]: IMO, "young" personifies the network too much, I think we should go with another option. What about:
```suggestion
          defaultMessage="You're joining a novel network. As with any new piece of software,
```

**Review by CarlBeek — APPROVED** [review]
There are dramatic improvements over the current version, well done to all! I have some thoughts mostly related to content below.

File: src/pages/Landing/SignupSteps/index.tsx [hunk 21]
```diff
@@ -31,36 +29,127 @@ const StepsContainer = styled.div`
...
+            emojiAlt={intl.formatMessage({ defaultMessage: 'clock' })}
+            title={intl.formatMessage({
+              defaultMessage: '6. Wait to become active',
+            })}
+            content={intl.formatMessage({
+              defaultMessage:
+                "Once set up, your validator won't become active straight away. Use this time to complete the checklist and get some extra practice on a testnet.",
+            })}
+          >
+            <Link to={routesEnum.checklistPage} primary>
```
[the remark this answers is not in the document]
Comment (wackerow) [reply src/pages/Landing/SignupSteps/index.tsx:149]: Tried this with the hash link but something is buggy where it's not moving to the hash link when the page loads. Could we bump this to a separate issue to follow instead?

File: src/pages/Acknowledgements/pageContent.tsx (hunk 20 above)
Comment (wackerow) [reply src/pages/Acknowledgements/pageContent.tsx]: Felt "novel" doesn't clearly convey the point here, which is that the network is still very new and could be buggy... tried using "You're joining an emerging, growing network." Let me know if you disagree

**Comment by wackerow** [comment]
image
Note, after "Mainnet" was changed to "mainnet" within `AppBar` we end up with (imo) a weird looking header bar. "mainnet" really feels like it should be capitalized in both places here as it's acting as a title.

File: src/pages/Landing/SignupSteps/index.tsx (hunk 21 above)
[the remark this answers is not in the document]
Comment (CarlBeek) [reply src/pages/Landing/SignupSteps/index.tsx:153]: Yeah, not critical. I was hoping it was an easy fix.

File: src/pages/Acknowledgements/pageContent.tsx (hunk 20 above)
Comment (CarlBeek) [reply src/pages/Acknowledgements/pageContent.tsx]: I don't think "emerging, growing" captures it either. The network has already emerged and eth1 is growing too. Some other options:
* pioneering
* trailblazing
* unprecedented

Comment (CarlBeek) [reply src/pages/Acknowledgements/pageContent.tsx]: I don't think "emerging, growing" captures it either. The network has already emerged and eth1 is growing too. Some other options:
* pioneering
* trailblazing
* unprecedented

Comment (ryancreatescopy) [reply src/pages/Acknowledgements/pageContent.tsx]: I think unprecedented works. Or we could reword it to "You're joining a network in its early stages." or "You're joining a type of network that's never existed before".

Comment (CarlBeek) [reply src/pages/Acknowledgements/pageContent.tsx]: I'm happy with either of those options.

**Comment by OGKEEKS** [comment]
if i wanted to help with this project by translating it to spanish would that be possible?
